ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/Runner.java

```diff
@@ -60,53 +61,55 @@
   private final HttpResponseFactory responseFactory = new HttpResponseFactory();
   private final Path dataDirectory;
 
-  private final Vertx vertx;
+  private Vertx vertx;
   private String jsonRpcHttpServiceId;
   private HttpServerService httpServerService;
 
   public Runner(
       final TransactionSerialiser serialiser,
-      final Vertx vertx,
       final HttpClientOptions clientOptions,
       final HttpServerOptions serverOptions,
       final Duration httpRequestTimeout,
       final TransactionFactory transactionFactory,
       final Path dataDirectory) {
     this.serialiser = serialiser;
     this.clientOptions = clientOptions;
+    LOG.debug("CLIENT OPTIONS: clientOptions.toString(): " + clientOptions.toString());
     this.serverOptions = serverOptions;
     this.httpRequestTimeout = httpRequestTimeout;
     this.transactionFactory = transactionFactory;
     this.dataDirectory = dataDirectory;
-    this.vertx = vertx;
   }
 
   public void start() {
+    this.vertx = Vertx.vertx();
     httpServerService = new HttpServerService(router(), serverOptions);
     vertx.deployVerticle(httpServerService, this::httpServerServiceDeployment);
   }
 
   public void stop() {
-    vertx.undeploy(jsonRpcHttpServiceId);
+    if (vertx != null) {
```

Comment:
by making the vertx always available, this conditional can go

Reply:
If something went wrong in the start then vertx might be null and you wouldn't want to throw a NullPointerException. This is good defensive programming ...